AEPMessaging/Tests/UnitTests/BatchedPropositionInteractionTests.swift

```diff
@@ -0,0 +1,225 @@
+/*
+ Copyright 2025 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import XCTest
+import AEPCore
+@testable import AEPMessaging
+import AEPTestUtils
+
+class BatchedPropositionInteractionTests: XCTestCase {
+    
+    // Common test properties
+    var scopeDetails1: [String: Any]!
+    var scopeDetails2: [String: Any]!
+    var proposition1: MockProposition!
+    var proposition2: MockProposition!
+    var propositionItem1: PropositionItem!
+    var propositionItem2: PropositionItem!
+
+    
+    override func setUp() {
+        super.setUp()
+        
+        // Setup common test data
+        scopeDetails1 = ["key1": "value1"]
+        scopeDetails2 = ["key2": "value2"]
+        propositionItem1 = PropositionItem(itemId: "item1", schema: .contentCard, itemData: [:])
+        propositionItem2 = PropositionItem(itemId: "item2", schema: .contentCard, itemData: [:])        
+        proposition1 = MockProposition(uniqueId: "proposition1", scope: "scope1", scopeDetails: scopeDetails1, items: [propositionItem1])
+        proposition2 = MockProposition(uniqueId: "proposition2", scope: "scope2", scopeDetails: scopeDetails2, items: [propositionItem2])
+
+        propositionItem1.proposition = proposition1
+        propositionItem2.proposition = proposition2
+        
+    }
+    
+    func testBatchPropositionInteractionXDM_forDisplayEvent() {
+        // setup
+        let propositionItems = [propositionItem1!, propositionItem2!]
```

Comment:
i guess you don't need `!` for these items since they are declared that way above